build.gradle.kts

```diff
@@ -241,16 +241,12 @@ tasks.register("javaPreCommit") {
   dependsOn(":runners:core-java:build")
   dependsOn(":runners:direct-java:build")
   dependsOn(":runners:extensions-java:metrics:build")
-  dependsOn(":runners:flink:1.12:build")
-  dependsOn(":runners:flink:1.12:job-server:build")
-  dependsOn(":runners:flink:1.13:build")
-  dependsOn(":runners:flink:1.13:job-server:build")
-  dependsOn(":runners:flink:1.14:build")
-  dependsOn(":runners:flink:1.14:job-server:build")
-  dependsOn(":runners:flink:1.15:build")
-  dependsOn(":runners:flink:1.15:job-server:build")
-  dependsOn(":runners:flink:1.16:build")
-  dependsOn(":runners:flink:1.16:job-server:build")
+  // lowest and highest supported flink versions
+  var flinkVersions = project.ext.get("allFlinkVersions") as Array<*>
+  dependsOn(":runners:flink:${flinkVersions[0]}:build")
+  dependsOn(":runners:flink:${flinkVersions[0]}:job-server:build")
+  dependsOn(":runners:flink:${flinkVersions[flinkVersions.size - 1]}:build")
+  dependsOn(":runners:flink:${flinkVersions[flinkVersions.size - 1]}:job-server:build")
```

Comment:
How often are we aware of catching a java issue with flink specific tests? I'd probably vote we just move all versions into the separate suite where it can run only on flink changes. In general, I'm a big fan of smaller more targeted suites to reduce flakiness.

If the flink tests would've caught a real issue, we'll still get that when the action does its normal scheduled run.

Reply:
> How often are we aware of catching a java issue with flink specific tests?

I do not remember a specific example, but I imagine this could happen in general.

We may still need at least one exercised in the main test suite, for the purpose of code coverage stat.

Similar practice is done in Spark runner precommit. In main test suite it runs with Spark 3.2; there is an Spar3_Version Precommit run additonal versions: 

https://github.com/apache/beam/blob/a4f29348ff7eadf8e0330e28815fdc30a53e0f0e/runners/spark/3/build.gradle#L35 (btw I should name the test suite "job_PreCommit_Java_Flink_Versions" in alignment with Spark version)

Will make the main precommit suite run only one flink version, and rename the test

Reply:
Ok; my vote is to split it out entirely, but I can live with just one suite

> We may still need at least one exercised in the main test suite, for the purpose of code coverage stat.

I don't think this is a good reason to make decisions like this. Code coverage is just a number that can be a useful indicator of places we have gaps, if its keeping us from making optimal decisions about our infrastructure we should ignore it.